File: src/MixNetFx/FAC_Generic/FacessoExceptions/FacessoAppExceptionBase.vb
Public MustInherit Class FacessoApplicationExceptionBase
    Inherits ApplicationException

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.New(Message, innerException)
    End Sub

End Class

Public Class FacessoGenericApplicationException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, InnerException)
    End Sub

End Class

Public Class FacessoLoginException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, InnerException)
    End Sub

End Class

Public Class FacessoEndOfSetupException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, innerException)
    End Sub

End Class

Public Class FacessoSqlDbException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, InnerException)
    End Sub

End Class

Public Class FacessoLicenseViolationException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, InnerException)
    End Sub

End Class

Public Class FacessoUniqueFieldAlreadyExistsException
    Inherits FacessoApplicationExceptionBase

    Sub New()
    End Sub

    Sub New(ByVal Message As String, ByVal innerException As Exception)
        MyBase.new(Message, InnerException)
    End Sub

End Class

File: src/MixNetFx/Facesso.Generic/FacessoExceptions/FacessoAppExceptionBase.cs
using System;

namespace Facesso
{
    public abstract class FacessoApplicationExceptionBase : ApplicationException
    {
        protected FacessoApplicationExceptionBase() { }

        protected FacessoApplicationExceptionBase(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoGenericApplicationException : FacessoApplicationExceptionBase
    {
        public FacessoGenericApplicationException() { }

        public FacessoGenericApplicationException(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoLoginException : FacessoApplicationExceptionBase
    {
        public FacessoLoginException() { }

        public FacessoLoginException(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoEndOfSetupException : FacessoApplicationExceptionBase
    {
        public FacessoEndOfSetupException() { }

        public FacessoEndOfSetupException(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoSqlDbException : FacessoApplicationExceptionBase
    {
        public FacessoSqlDbException() { }

        public FacessoSqlDbException(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoLicenseViolationException : FacessoApplicationExceptionBase
    {
        public FacessoLicenseViolationException() { }

        public FacessoLicenseViolationException(string message, Exception innerException)
            : base(message, innerException) { }
    }

    public class FacessoUniqueFieldAlreadyExistsException : FacessoApplicationExceptionBase
    {
        public FacessoUniqueFieldAlreadyExistsException() { }

        public FacessoUniqueFieldAlreadyExistsException(string message, Exception innerException)
            : base(message, innerException) { }
    }
}
